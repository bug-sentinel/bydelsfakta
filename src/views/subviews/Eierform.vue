<template>
  <div class="main-container">
    <div class="main-container__cards">
      <h2 class="section-heading">Eierform</h2>
      <graph-card v-for="(card, index) in settings.eierform" :key="`eierform-${index}`" :settings="card" />
    </div>
    <div class="main-container__map">
      <h3>Kart</h3>
    </div>
  </div>
</template>

<script>
import GraphCard from '../../components/GraphCard.vue';

export default {
  name: 'Eierform',
  components: { GraphCard },
  props: {
    settings: {
      type: Object,
      required: true,
    },
  },
};
</script>

<style scoped></style>
