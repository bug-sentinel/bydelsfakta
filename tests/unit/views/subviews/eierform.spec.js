import { mount, createLocalVue } from '@vue/test-utils';
import VueResize from 'vue-resize';
import Eierform from '../../../../src/views/subviews/Eierform.vue';
import page from '../../../../src/config/pages';
import clickOutside from '../../../../src/directives/clickOutside';

describe('Eierform', () => {
  let wrapper = null;

  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(VueResize);
    localVue.directive('click-outside', clickOutside);
    wrapper = mount(Eierform, {
      propsData: {
        settings: page['eierform'],
      },
      localVue,
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('renders Eierform component correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
