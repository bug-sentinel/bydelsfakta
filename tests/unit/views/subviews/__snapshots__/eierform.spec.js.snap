// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Eierform renders Eierform component correctly 1`] = `
<div
  class="main-container"
>
  <div
    class="main-container__cards"
  >
    <h2
      class="section-heading"
    >
      Eierform
    </h2>
     
    <div
      class="card-container large"
    >
      <div
        class="card"
      >
        <div
          class="tabs"
          role="tablist"
        >
          <button
            aria-label="Status"
            aria-selected="[object Object]"
            class="tabs__button active"
            role="tab"
          >
            Status
          </button>
          <button
            aria-label="Matrise"
            aria-selected="[object Object]"
            class="tabs__button"
            role="tab"
          >
            Matrise
          </button>
          <button
            aria-label="Historisk"
            aria-selected="[object Object]"
            class="tabs__button"
            role="tab"
          >
            Historisk
          </button>
           
          <div
            class="context-menu"
            role="menu"
          >
            <button
              aria-haspopup="true"
              aria-label="Ã¥pne meny"
              class="context-menu__button"
            >
              <i
                class="material-icons"
              >
                menu
              </i>
            </button>
             
            <!---->
          </div>
        </div>
         
        <div
          class="graph__container"
        >
          <svg
            class="graph__svg"
            style="font-family: OsloSans;"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <text
              class="heading"
              font-size="14"
              font-weight="bold"
              y="14"
            />
            <g
              class="canvas"
              transform="translate(190, 100)"
            >
              <g
                class="axis x"
              />
              <g
                class="axis y"
              />
              <g
                class="axis x2"
              />
              <g
                class="axis y2"
              />
              <g
                class="columns"
              />
              <g
                class="rows"
              />
              <g
                class="close"
                style="display: none;"
              >
                <rect
                  fill="#FF8174"
                  height="30"
                  opacity="0.7"
                  style="cursor: pointer;"
                  width="30"
                />
                <text
                  fill="#292858"
                  font-size="20"
                  font-weight="700"
                  style="pointer-events: none;"
                  text-anchor="middle"
                  transform="translate(15, 20)"
                >
                  x
                </text>
              </g>
            </g>
            <g
              class="sourceGroup"
              fill="#292858"
              opacity="0"
            >
              <text
                class="source-label"
                font-size="10"
              >
                Kilde: 
              </text>
              <text
                class="source"
                font-size="10"
              />
            </g>
          </svg>
           
          <div
            class="resize-observer"
            data-v-b329ee4c=""
            tabindex="-1"
          >
            <object
              aria-hidden="true"
              data="about:blank"
              tabindex="-1"
              type="text/html"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="main-container__map"
  >
    <h3>
      Kart
    </h3>
  </div>
</div>
`;
